### PR TITLE
Supports new agreements data structure and IDM roles

### PR DESCRIPTION
**Pull request by james-siteclick**
WATER-2688
* Supports new shape of data returned from water service
* Ensures agreement descriptions are displayed on list view page
* Supports new IDM roles for managing/deleting agreements

### CODE REVIEW

File: src/internal/modules/agreements/routes.js [hunk 1]
```diff
@@ -2,8 +2,8 @@ const controller = require('./controller');
 const preHandlers = require('./pre-handlers');
 const { VALID_GUID } = require('shared/lib/validators');
 
-const { charging } = require('internal/lib/constants').scope;
-const allowedScopes = [charging];
+const { deleteAgreements } = require('internal/lib/constants').scope;
+const allowedScopes = [deleteAgreements];
```
Comment (timtamimi) [inline src/internal/modules/agreements/routes.js:6]: Should this be `manageAgreements`?

File: src/internal/views/nunjucks/agreements/delete.njk [hunk 2]
```diff
@@ -14,10 +14,14 @@
   <div class="phone--show">
     <hr class="govuk-section-break govuk-section-break--m govuk-section-break--visible">
 
-    <p>Agreement: {{ agreement.agreement.description }}</p>
-    <p>Date signed: {{ agreement.dateSigned | date }}</p>
-    <p>Start date: {{ agreement.dateRange.startDate | date }}</p>
-    <p>End date: {{ agreement.dateRange.endDate | date }}</p>
+    <p>Agreement:
+      {{ agreement.agreement.description }}</p>
```
Comment (timtamimi) [inline src/internal/views/nunjucks/agreements/delete.njk:18]: Is the agreement description always populated? If not, might it be worth adding a fallback text? 

Ignore if not.

File: src/internal/views/nunjucks/view-licences/tabs/charge.njk [hunk 3]
```diff
@@ -14,19 +14,23 @@
...
 
-    <p>Start date: {{ chargeVersion.dateRange.startDate | date }}</p>
-    <p>End date: {{ chargeVersion.dateRange.endDate | date }}</p>
-    <p>Reason: </p>
-    <p>Status: {{ badge(chargeVersion | chargeVersionBadge) }}</p>
+    <p>Start date:
+      {{ chargeVersion.dateRange.startDate | date }}</p>
+    <p>End date:
+      {{ chargeVersion.dateRange.endDate | date }}</p>
+    <p>Reason:
```
Comment (timtamimi) [inline src/internal/views/nunjucks/view-licences/tabs/charge.njk:30]: Reason here doesn't have a value. Ignore if this is intentional.

**Review by timtamimi — COMMENTED** [review]
Please refer to my pedantic questions. I think they're all false alarms, but I thought I should check first.

File: src/internal/modules/agreements/routes.js (hunk 1 above)
Comment (james-siteclick) [reply src/internal/modules/agreements/routes.js:6]: deleteAgreements is OK here as the two controller actions there are for deleting

File: src/internal/views/nunjucks/agreements/delete.njk (hunk 2 above)
Comment (james-siteclick) [reply src/internal/views/nunjucks/agreements/delete.njk:18]: I've added the code as a fallback thanks

File: src/internal/views/nunjucks/view-licences/tabs/charge.njk (hunk 3 above)
Comment (james-siteclick) [reply src/internal/views/nunjucks/view-licences/tabs/charge.njk:30]: I've added this to ticket 2845 which is the charge information wrap-up

**Review by timtamimi — APPROVED** [review]
Noice. Thanks for reviewing my comments.

**Pull request merged**